extension/encoding/awslogsencodingextension/internal/unmarshaler/vpc-flow-log/unmarshaler.go

```diff
@@ -78,91 +103,116 @@ func (v *vpcFlowLogUnmarshaler) UnmarshalAWSLogs(reader io.Reader) (plog.Logs, e
 	}
 }
 
-// resourceKey stores the account id and region
-// of the flow logs. All log lines inside the
-// same S3 file come from the same account and
-// region.
-//
-// See https://docs.aws.amazon.com/vpc/latest/userguide/flow-logs-s3-path.html.
-type resourceKey struct {
-	accountID string
-	region    string
-}
-
 func (v *vpcFlowLogUnmarshaler) unmarshalPlainTextLogs(reader io.Reader) (plog.Logs, error) {
-	scanner := bufio.NewScanner(reader)
+	// use buffered reader for efficiency and to avoid any size restrictions
+	bufReader := bufio.NewReader(reader)
 
-	var fields []string
-	if scanner.Scan() {
-		firstLine := scanner.Text()
-		fields = strings.Split(firstLine, " ")
+	b, err := bufReader.ReadByte()
+	if err != nil {
+		return plog.Logs{}, fmt.Errorf("failed to read first byte: %w", err)
 	}
 
+	err = bufReader.UnreadByte()
+	if err != nil {
+		return plog.Logs{}, fmt.Errorf("failed to unread first byte: %w", err)
+	}
+
+	if b == '{' {
+		// Read all bytes and forward to fromCloudWatch
+		line, err := io.ReadAll(bufReader)
+		if err != nil {
+			return plog.Logs{}, fmt.Errorf("failed to read cloudwatch bound VPC flow log: %w", err)
+		}
+
+		// Dealing with a JSON logs, so check for CW bound trigger
+		return v.fromCloudWatch(v.cfg.parsedFormat, line)
```

Comment:
Why do we read everything into memory here? fromCloudWatch is creating a reader out of it again

Reply:
sorry yes my bad, thanks for noticing and pointing this

Fixed with commit https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/44771/changes/0c6fd7b06acefe8d42e707164b00123d452efa97